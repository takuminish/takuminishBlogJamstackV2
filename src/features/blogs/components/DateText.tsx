import Typography from "@mui/material/Typography";
import { formatInTimeZone } from "date-fns-tz";
import React from "react";

export type Props = {
  label: string;
  date: Date;
};

function DateText(props: Props) {
  const { label, date } = props;

  return (
    <Typography variant="body2" color="text.secondary">
      {label}: {formatInTimeZone(date, "Asia/Tokyo", "yyyy/MM/dd")}
    </Typography>
  );
}

export default DateText;
