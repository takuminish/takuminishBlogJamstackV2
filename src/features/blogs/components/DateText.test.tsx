import React from "react";
import { render, screen } from "@testing-library/react";
import DateText, { Props } from "./DateText";

const createDOM = (props?: Props) => {
  const defaultProps: Props = {
    label: "公開日",
    date: new Date("2021-06-12T15:00:00.000Z"),
  };

  const { label, date } = props ?? defaultProps;

  return render(<DateText label={label} date={date} />);
};

test("snapshot", () => {
  const { container } = createDOM();

  expect(container).toMatchSnapshot();
});

test("propsで与えられたlabelが表示されていること", async () => {
  createDOM();

  await screen.findByText(/公開日/);
});

test("propsで与えられたldateがyyyy/MM/dd形式で表示されること", async () => {
  createDOM();

  await screen.findByText(/2021[/]06[/]13/);
});
