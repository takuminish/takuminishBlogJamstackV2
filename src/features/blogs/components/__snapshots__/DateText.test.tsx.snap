// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot 1`] = `
<div>
  <p
    class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
  >
    公開日
    : 
    2021/06/13
  </p>
</div>
`;
